frontends/web/src/routes/account/summary/accountssummary.tsx

```diff
@@ -15,199 +15,193 @@
   ) => {
-    this.setState(({ syncStatus }) => {
-      return {
-        syncStatus: {
-          ...syncStatus,
-          [code]: syncedAddressesCount,
-        }
-      };
-    });
-  };
+    setSyncStatus((prevSyncStatus) => ({
+      ...prevSyncStatus,
+      [code]: syncedAddressesCount,
+    }));
+  }, []);
 
-  private onEvent = (payload: TPayload) => {
+  const onStatusChanged = useCallback(async (code: string, initial: boolean = false) => {
+    const status = await accountApi.getStatus(code);
+    if (status.disabled) {
+      return;
+    }
+    if (!status.synced) {
+      return accountApi.init(code);
+    }
+    const balance = await accountApi.getBalance(code);
+    setBalances((prevBalances) => ({
+      ...prevBalances,
+      [code]: balance
+    }));
+    if (initial) {
+      return;
+    }
+    getAccountsTotalBalance();
+  }, []);
+
+  const onEvent = useCallback((payload: TPayload) => {
     if ('type' in payload) {
       const { code, data, type } = payload;
       if (type === 'account') {
         switch (data) {
         case 'statusChanged':
         case 'syncdone':
           if (code) {
-            this.onStatusChanged(code);
+            onStatusChanged(code);
           }
-          // Force getting account summary now; cancel next scheduled call.
-          window.clearTimeout(this.summaryReqTimerID);
-          this.summaryReqTimerID = undefined;
-          this.getAccountSummary();
+          getAccountSummary();
           break;
         }
       }
     }
+  }, [onStatusChanged]);
+
+  const subscribe = useCallback(() => {
+    unsubscribe(unsubscribeList.current);
+    unsubscribeList.current.push(apiWebsocket(onEvent));
+    accounts.forEach(account => {
+      unsubscribeList.current.push(syncAddressesCount(account.code, onSyncAddressesCount));
+      onStatusChanged(account.code, firstRender.current);
+    });
+  }, [onSyncAddressesCount, onStatusChanged, onEvent, accounts]);
+
+  const exportSummary = () => {
+    accountApi.exportSummary().then(setExported).catch(console.error);
   };
 
-  private async onStatusChanged(code: string, initial: boolean = false) {
-    const status = await accountApi.getStatus(code);
-    if (status.disabled) {
-      return;
-    }
-    if (!status.synced) {
-      return accountApi.init(code);
+  // fetch accounts summary and balance on the first render.
+  useEffect(() => {
+    if (firstRender.current) {
+      getAccountSummary();
+      getAccountsTotalBalance();
+      return () => {
+        firstRender.current = false;
+      };
     }
-    const balance = await accountApi.getBalance(code);
-    this.setState(({ balances }) => ({
-      balances: {
-        ...balances,
-        [code]: balance
-      }
-    }));
+  });
 
-    if (initial) {
-      return;
+  // update the timer to get a new account summary update when receiving the previous call result.
+  useEffect(() => {
+    // replace previous timer if present
+    if (summaryReqTimerID.current) {
```

Comment:
Needed to change a bit the timer handling, since the previous implementation didn't seem to be directly usable in a functional component. React needs the `useRef` hook to be able to update the timer in the `useEffect` (which was previously done in the callback related to the data status update).

Reply:
I am thinking if the clearTimeout should be instead in the return function of this useEffect, so that it clears on unmount, but also when summaryData changes instead of here.

and then you could remove this;

https://github.com/digitalbitbox/bitbox-wallet-app/pull/2131/files?diff=split&w=0#diff-e5d06f77122fc6a06ea5e7c80d3466cd0fa4d69a252e8108de78062a5bdbea83R159

Reply:
Very nice! Done :+1: